tests/unit/source_code/test_functional.py

```diff
@@ -47,7 +47,7 @@ def __init__(self, path: Path):
 
     def verify(self):
         expected_problems = list(self._expected_problems())
-        actual_problems = list(self._lint())
+        actual_problems = sorted(list(self._lint()), key=lambda a: (a.start_line, a.start_col))
```

Comment:
Sometimes we generate more than one piece of advice with the same starting position. To be stable I think we need to:
1. Include the `code` and the `message` in the sorting key.
2. Also sort the expected problems by the same criteria (although this is less important because the order in the samples can be adjusted.)